### PR TITLE
[HDInsight] Migrate to track2 python sdk 7.0.0

### PR DESCRIPTION
**Pull request by aim-for-better**
**Description**

This PR migrate az hdinsight module to track2 python sdk
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
HDInsight

File: src/azure-cli/azure/cli/command_modules/hdinsight/custom.py [hunk 1]
```diff
@@ -396,9 +397,9 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
     )
 
     if no_wait:
-        return sdk_no_wait(no_wait, client.create, resource_group_name, cluster_name, create_params)
+        return sdk_no_wait(no_wait, client.begin_create, resource_group_name, cluster_name, create_params)
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/hdinsight/custom.py:400]: This issue is not related to track 2 migration:
We just need `return sdk_no_wait(no_wait, client.begin_create, resource_group_name, cluster_name, create_params)`, no need to check `if no_wait` as `sdk_no_wait()` will handle this logic based on the value of `no_wait`.

Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/custom.py]: removed the duplicate check logic, thanks

**Review by jsntcy — APPROVED** [review]

**Comment by aim-for-better** [comment]
The CLI style failure is not related with HDInsight
The error is as bellow:
src/azure-cli/azure/cli/command_modules/resource/_win_vt.py:50:15: E1131: unsupported operand type(s) for | (unsupported-binary-operation)
src/azure-cli/azure/cli/command_modules/resource/_win_vt.py:50:15: E1131: unsupported operand type(s) for | (unsupported-binary-operation)

**Review by kairu-ms — APPROVED** [review]
